### PR TITLE
Fixes #1959, #1960, #1961 - Changes for robust error handling and reporting in docker-push

### PR DESCRIPTION
**Pull request by lake-of-dreams**
Fixes [wercker/backlog#1959](https://github.com/wercker/backlog/issues/1959), [wercker/backlog#1960](https://github.com/wercker/backlog/issues/1960), [wercker/backlog#1961 ](https://github.com/wercker/backlog/issues/1961)

- Change step.InitEnv to be able to return an error in case of any initialization errors
- Added runner_test.go as unit test for InitEnv change
- Removed log.panic from few places in InitEnv and return the error instead
- Fixed docker.InferRegistryAndRepository docs, additional error handing and better error messages while evaluating registry and repository
- Enhanced error message from docker.tagAndPush for errors originating from docker daemon while doing a push
- Added test case for invalid registry and repository checking for InferrEgistryAndRepository
- vendor.json points to [issue-1960](https://github.com/wercker/docker-check-access/tree/issue-1960) branch of docker-check-access which contains changes mentioned in PR [wercker/docker-check-access#PR9](https://github.com/wercker/docker-check-access/pull/9). Will point back to master after merging

### CODE REVIEW

**Comment by lake-of-dreams** [comment]
Have been tested using canary

File: docker/docker.go [hunk 1]
```diff
@@ -750,15 +753,15 @@ func InferRegistryAndRepository(repository string, registry string, pipelineOpti
 	}
 
 	if len(strings.TrimSpace(inferredRegistry)) != 0 {
-		regsitryURLFromStepConfig, err := url.Parse(inferredRegistry)
+		regsitryURLFromStepConfig, err := url.ParseRequestURI(inferredRegistry)
```
Comment (nigeldeakin) [inline docker/docker.go]: Would be nice to correct the (existing) typo on `regsitryURLFromStepConfig` (to `registryURLFromStepConfig`)

Comment (nigeldeakin) [reply docker/docker.go]: Can you please explain why `Parse` wasn't correct here? (This is for my benefit: I read the docs but it was unclear).

Comment (lake-of-dreams) [reply docker/docker.go]: Fixed Typo. url.Parse() also allows relative paths - e.g. a simple word "registry" will also be valid/ParseRequestURI  requires scheme (http, https) or absolute paths and therefore a stricter check (scheme will anyways be added by normalizeRegistry which will be called before this method but with ParseRequestURI it becomes a stricter check at this method level itself)

**Pull request merged**